docusaurus/docs/create-a-plugin/extend-a-plugin/fetch-data-from-frontend.md

```diff
@@ -0,0 +1,287 @@
+
+Here's an example of how your configuration form might look like
+
+```typescript title="src/ConfigEditor.tsx"
+export function ConfigEditor(props: Props) {
+  const { onOptionsChange, options } = props;
+  const { jsonData } = options;
+
+  const onApiUrlChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onOptionsChange({
+      ...options,
+      jsonData: {
+        ...jsonData,
+        // notice we set the apiUrl value inside jsonData
+        apiUrl: event.target.value,
+      },
+    });
+  };
+
+  return (
+    <InlineField label="apiUrl" labelWidth={12}>
+      <Input
+        onChange={onApiUrlChange}
+        value={jsonData.apiUrl || ''}
+        placeholder="json field returned to frontend"
+        width={40}
+      />
+    </InlineField>
+  );
+}
+```
+
+Once the field is set correctly, you may use it inside your data source. Following the previous example for the data source code, you can now simply use it as this:
+
+```typescript
+const response = getBackendSrv().fetch<TODO[]>({
+  // Notice we use `interpolation` as defined
+  // in the route path.
+  // See the previous examples to see where
+  // this.baseUrl comes from.
+  url: `${this.baseUrl}/interpolation`,
+});
+```
+
+## Using other HTTP Protocols e.g. POST, PUT, DELETE with the data proxy
+
+You can specify the method directly in the `fetch` method. Your routes in `src/plugin.json` remain the same:
+
+```typescript
+const response = getBackendSrv().fetch<TODO[]>({
+  url: `${this.baseUrl}`,
+  method: 'POST',
+  data: dataToSendViaPost,
+});
+```
+
+## Adding authentication to your requests using the data proxy
+
+You can read about adding authentication to the data proxy [here](./add-authentication-for-data-source-plugins.md)
+
+## Debugging requests from the data proxy
+
+If you wish to debug the requests that are going from the Grafana backend to your API, you can enable the data proxy logging in the [configuration](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#dataproxy)
+
+You must also [enable debug logs](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#mode) in Grafana to be able to see the dataproxy logs
+
+In your Grafana configuration file:
+
+```
+[log]
+level = debug
+
+[dataproxy]
+logging = true
+```
+
+Now you will be able to see in the Grafana server output the requests going out to your API from the data proxy.
+
+# FAQ
+
+## I am using the [[DataSourceHttpSettings](https://developers.grafana.com/ui/latest/index.html?path=/story/data-source-datasourcehttpsettings--basic)](https://developers.grafana.com/ui/latest/index.html?path=/story/data-source-datasourcehttpsettings--basic) component. Where's the URL from the user?
+
+The DataSourceHttpSettings sets the URL directly in the options that it is stored in `instanceSettings.url` instead of inside the jsonData. You can query that URL directly.
```

Comment:
Might want to point out that no routes might be needed in this case. 

Wonder if using the HTTP settings component should be discussed higher up as the first example since we want to encourage the usage of that. Thoughts?